department-of-veterans-affairs/caseflow
# HunJerBAH/APPEALS-7497 Feature Toggle Update

Resolves #{jira issue link}

# Description
Updated MST/PACT feature toggles to restrict workflow if the feature toggles are turned off. 

## Acceptance Criteria
- [ ] MST/PACT workflow works as intended with the feature toggles on
- [ ] MST/PACT workflow is not visible if the feature toggles are turned off.

## Testing Plan
1. Open up rails console and turn off the feature toggles by running:
`FeatureToggle.disable!(:mst_identification)`
`FeatureToggle.disable!(:pact_identification)`
`FeatureToggle.disable!(:legacy_mst_pact_identification)`

2. Sign in as BVADWISE (or another intake user) and navigate through Intake. Note the MST/PACT checkboxes are not showing and the special designation isn't showing.
3. Navigate to the Case Details page. Notice the MST/PACT special issue designation isn't showing and there are no IssueUpdate tasks.
4. Click on the Correct Issues link. Note the MST/PACT checkboxes are not showing, the special designation isn't showing and the Edit issue dropdown action is not an option.
5. Find a Legacy appeal and navigate to the Case Details page. Notice how the special issue designation isn't showing and there is no way to access the correct issues link.
6. Sign in as BVAAABSHIRE and navigate to the Queue. Select a Legacy case and navigate to the Case Details page. Click on the action dropdown to navigate to the Judge workflow. Notice how there is no special issue designation. Click on **edit issue**. Notice on the page how there is no MST/PACT checkboxes
7. Go back to the Judge's Queue and select an AMA case. Navigate to the Add Decisions workflow. On the Add Decisions page, notice how there are no special issue designations. Click on **edit issue**. Notice how there are no MST/PACT checkboxes.  


# Frontend
## User Facing Changes

 BEFORE (feature toggle on for MST/PACT)
![image](https://github.com/department-of-veterans-affairs/caseflow/assets/99915461/ee742a53-ecc9-46c3-be39-78e2aad9167a)
![image](https://github.com/department-of-veterans-affairs/caseflow/assets/99915461/53e5fc8a-ad6a-402a-b0bf-743e5f1b5b59)
![image](https://github.com/department-of-veterans-affairs/caseflow/assets/99915461/2a1a72db-a015-409a-ab4e-1e2d79b9cd73)
![image](https://github.com/department-of-veterans-affairs/caseflow/assets/99915461/70a05e34-c17d-4b37-8dad-99564fce4b90)

![image](https://github.com/department-of-veterans-affairs/caseflow/assets/99915461/1628db42-86cf-45d9-b19c-dbf39875efbc)

![image](https://github.com/department-of-veterans-affairs/caseflow/assets/99915461/e8a6a9c4-2245-4c31-b5f8-e69f46927fe5)

![image](https://github.com/department-of-veterans-affairs/caseflow/assets/99915461/9a391e35-8eab-4764-98b6-8829eb68b5d7)

![image](https://github.com/department-of-veterans-affairs/caseflow/assets/99915461/1136d87a-768b-494f-a437-ec4820bc7506)

![image](https://github.com/department-of-veterans-affairs/caseflow/assets/99915461/4b9e3e0d-c63c-4282-b033-6bf7d8e946c4)

![image](https://github.com/department-of-veterans-affairs/caseflow/assets/99915461/bf009331-79be-41e0-8053-278cb5bc4e73)

![image](https://github.com/department-of-veterans-affairs/caseflow/assets/99915461/cfbb045b-83b5-4521-8263-c2b73601bee9)

**AFTER** (feature toggles turned off)

**Intake**
![image](https://github.com/department-of-veterans-affairs/caseflow/assets/99915461/294068fc-e27e-4788-b931-ef5a0bcf9b12)

![image](https://github.com/department-of-veterans-affairs/caseflow/assets/99915461/b9537f29-7c16-4b5e-98f2-1bc4e1bd33e3)

![image](https://github.com/department-of-veterans-affairs/caseflow/assets/99915461/43075df1-2802-4c60-a13f-36c99a4d1af9)

![image](https://github.com/department-of-veterans-affairs/caseflow/assets/99915461/491ff02f-031b-405e-ada3-2acf3eb36302)

**Queue**
![image](https://github.com/department-of-veterans-affairs/caseflow/assets/99915461/01eabda1-5114-46dd-b6c0-c13d9d8b82c2)

![image](https://github.com/department-of-veterans-affairs/caseflow/assets/99915461/4f290310-2500-4060-8268-bd4d6b5c3653)

![image](https://github.com/department-of-veterans-affairs/caseflow/assets/99915461/8deee2de-eccb-495c-b3a3-1a95f674ef65)

**Decisions Workflow**
![image](https://github.com/department-of-veterans-affairs/caseflow/assets/99915461/025c5f51-950e-49d2-8ff8-21e07d0cf3c5)

![image](https://github.com/department-of-veterans-affairs/caseflow/assets/99915461/a151f324-db88-4147-8be4-11cd1afeca81)

**Legacy Decisions Workflow**
![image](https://github.com/department-of-veterans-affairs/caseflow/assets/99915461/f0543a73-3626-4a0b-9756-d1faf59472c6)

![image](https://github.com/department-of-veterans-affairs/caseflow/assets/99915461/1d16854d-641c-43c9-be35-e8aa1bf63805)
